docs/proposals/config_data.md

```diff
@@ -0,0 +1,316 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named `ConfigData` stored inside `etcd` that
```

Comment:
This isn't inherently tied to etcd. It's tied to whatever store Kubernetes uses, which is currently etcd, regardless whether it's a separate instance or not.

#1957 is the issue about supporting other stores, such as Consul.